File: tkpdfeed/src/main/java/com/tokopedia/tkpd/tkpdfeed/feedplus/view/data/repository/FeedRepository.java
package com.tokopedia.tkpd.tkpdfeed.feedplus.view.data.repository;

import com.tokopedia.core.base.domain.RequestParams;
import com.tokopedia.tkpd.tkpdfeed.feedplus.view.domain.model.DataFeedDomain;
import com.tokopedia.tkpd.tkpdfeed.feedplus.view.domain.model.feeddetail.DataFeedDetailDomain;
import com.tokopedia.tkpd.tkpdfeed.feedplus.view.domain.model.FeedResult;

import java.util.List;

import rx.Observable;

/**
 * @author ricoharisin .
 */

public interface FeedRepository {

    Observable<FeedResult> getFeedsFromCloud(RequestParams requestParams);

    Observable<FeedResult> getFirstPageFeedsFromCloud();

    Observable<FeedResult> getFirstPageFeedsFromLocal();

    Observable<List<DataFeedDetailDomain>> getFeedsDetail(RequestParams requestParams);


}

File: features/content/feed/src/main/java/com/tokopedia/feedplus/domain/repository/FeedRepository.kt
package com.tokopedia.feedplus.domain.repository

import com.tokopedia.feedplus.presentation.model.FeedTabsModel
import com.tokopedia.feedplus.presentation.model.MetaModel

/**
 * Created by meyta.taliti on 08/08/23.
 */
interface FeedRepository {

    suspend fun getTabs(): FeedTabsModel

    suspend fun getMeta(): MetaModel
}
